esda/join_counts.py

```diff
@@ -143,22 +177,32 @@ def __init__(self, y, w, permutations=PERMUTATIONS):
             self.min_bw = np.min(self.sim_bw)
             self.mean_bw = np.mean(self.sim_bw)
             self.max_bw = np.max(self.sim_bw)
+            self.sim_chi2 = sim_jc[:, 3]
             p_sim_bb = self.__pseudop(self.sim_bb, self.bb)
             p_sim_bw = self.__pseudop(self.sim_bw, self.bw)
+            p_sim_chi2 = self.__pseudop(self.sim_chi2, self.chi2)
             self.p_sim_bb = p_sim_bb
             self.p_sim_bw = p_sim_bw
+            self.p_sim_chi2 = p_sim_chi2
 
     def __calc(self, z):
-        zl = lag_spatial(self.w, z)
-        bb = sum(z * zl) / 2.0
-        zw = 1 - z
-        zl = lag_spatial(self.w, zw)
-        ww = sum(zw * zl) / 2.0
-        bw = self.J - (bb + ww)
-        return (bb, ww, bw)
+        adj_list = self.adj_list
+        focal = z[adj_list.focal]
```

Comment:
I think this needs the following to account for string-indexed weights:
```python
zseries = pandas.Series(z, index=self.w.id_order)
focal = zseries.loc[adj_list.focal]
neighbor = zseries.loc[adj_list.neighbor]
```
I can't test this without pysal/libpysal#166